ecs-init/docker/docker.go

```diff
@@ -498,6 +547,37 @@ func getCapabilityExecBinds() []string {
 	return binds
 }
 
+// isDomainJoined is used to validate if container instance is part of a valid active directory.
+func isDomainJoined() bool {
+	realmPathByte, err := exec.Command("bash", "-c", "which realm").Output()
+	if err != nil {
+		log.Errorf("realmd is not installed on the instance")
+		return false
+	}
+
+	realmPathString := strings.TrimSpace(string(realmPathByte))
+	if len(realmPathString) == 0 {
+		log.Errorf("could not path of realm on the instance")
+		return false
+	}
+
+	realmCommmand := realmPathString + " list | grep 'realm-name'"
+	realmName, err := exec.Command("bash", "-c", realmCommmand).Output()
```

Comment:
Why does this have to run inside a bash shell? Also prefer that you run `realm list` and then do the string parsing in golang, rather than piping the command to grep.

Reply:
addressed suggestion - https://github.com/aws/amazon-ecs-agent/pull/3445 (as per meeting making this PR into smaller chunks)